### PR TITLE
Fixes regzo2/PicoStreamingAssistantFTUDP#16

### PR DESCRIPTION
**Pull request by lonelyicer**

### CODE REVIEW

**Comment by lonelyicer** [comment]
Fixes regzo2/PicoStreamingAssistantFTUDP#16

**Comment by lonelyicer** [comment]
Fixes regzo2/PicoStreamingAssistantFTUDP#16

File: PicoStreamingAssistantFTUDP/Pico4SAFTExtTrackingModule.cs [hunk 1]
```diff
@@ -25,23 +25,37 @@ public sealed class Pico4SAFTExtTrackingModule : ExtTrackingModule, IDisposable
...
 
-        Logger.LogInformation("Using {}.", this.connector.GetProcessName());
-        return true;
+            Logger.LogInformation("Using {}.", this.connector.GetProcessName());
+            return true;
+        }
+        else
+        {
+            var programChecker = new ProcessRunningProgramChecker();
+            bool using_sa = programChecker.Check(PicoPrograms.StreamingAssistant);
```
Comment (rogermiranda1000) [inline PicoStreamingAssistantFTUDP/Pico4SAFTExtTrackingModule.cs:45]: This basically is code duplication of ConnectorFactory, and I'd rather not do that. Also, at first sight I don't see a difference between both branches... I'll have to check exactly why this solves the issue

**Review by rogermiranda1000 — CHANGES_REQUESTED** [review]
This fix **won't** work; check my notes on https://github.com/regzo2/PicoStreamingAssistantFTUDP/issues/16#issuecomment-2038183605 . I'll provide a fix myself tomorrow, in case you want to wait for it. If you update this PR before my fix I'll take a look at it. Thanks for finding the bug!

**Comment by lonelyicer** [comment]
> This fix **won't** work; check my notes on [#16 (comment)](https://github.com/regzo2/PicoStreamingAssistantFTUDP/issues/16#issuecomment-2038183605) . I'll provide a fix myself tomorrow, in case you want to wait for it. If you update this PR before my fix I'll take a look at it. Thanks for finding the bug!

I did this because, after doing so, although the timeout problem still exists, it still keeps the data transferring normally, and the tracking is not a problem

**Comment by lonelyicer** [comment]
> > This fix **won't** work; check my notes on [#16 (comment)](https://github.com/regzo2/PicoStreamingAssistantFTUDP/issues/16#issuecomment-2038183605) . I'll provide a fix myself tomorrow, in case you want to wait for it. If you update this PR before my fix I'll take a look at it. Thanks for finding the bug!
> 
> I did this because, after doing so, although the timeout problem still exists, it still keeps the data transferring normally, and the tracking is not a problem

But the reinitialize module looks like a better solution

**Comment by rogermiranda1000** [comment]
> I did this because, after doing so, although the timeout problem still exists, it still keeps the data transferring normally

Oh, I didn't expect that... maybe the issue is that the socket fails one time but the socket is still active...

**Review by rogermiranda1000 — CHANGES_REQUESTED** [review]
Looks like this changes solves the issue about entering&exiting sleep mode and preventing VRCFT to be rebooted, and also it looks like the "Please ensure one program is running to send tracking data" check is no longer needed with this fix.

However, I did notice another bug more problematic (in both Business Streaming and PICO Connect):
If you launch the app, connect to SteamVR, and launch VRCFT you'll see how the module starts. Now, remove the headset and let it enter sleep mode; the added code will be triggered ("Waiting for data..."). Now, (without exiting the headset sleep mode) close VRCFT. It won't close and report as "not responding".
I'll try to see who's causing this issue (most likely the `Dispose` method), but I won't merge the PR until this issue gets fixed.

File: PicoStreamingAssistantFTUDP/Pico4SAFTExtTrackingModule.cs [hunk 2]
```diff
@@ -213,9 +213,11 @@ public override void Update()
         }
         catch (SocketException ex) when (ex.ErrorCode is 10060)
         {
-            if (!StreamerValidity())
-                Logger.LogInformation("Streaming Assistant, Business Streaming or PICO Connect is currently not running. Please ensure one program is running to send tracking data.");
```
[the remark this answers is not in the document]
Comment (rogermiranda1000) [reply PicoStreamingAssistantFTUDP/Pico4SAFTExtTrackingModule.cs:217]: After some testing I've concluded that this part can be deleted without any problem.

**Comment by rogermiranda1000** [comment]
Here's why it gets stuck: the `Initialize()` call ( https://github.com/regzo2/PicoStreamingAssistantFTUDP/pull/17/files#diff-6e070d01f5c9e071f65b958859432bceaa872dfc097100051e5414f9641afcc9R219 ) blocks the execution, and VRCFT won't call `Dispose()` until the `Update` exits its execution. I'll close this PR as I've finally decided to move the responsability of reinitializing the connection on the `LegacyConnector`, the PR was https://github.com/regzo2/PicoStreamingAssistantFTUDP/pull/19

**Pull request closed without merging**